src/main/java/uk/gov/ons/ctp/response/action/service/ActionProcessingService.java

```diff
@@ -167,10 +167,15 @@ private ActionCancel prepareActionCancel(final Action action) {
    * @param event the event to transition the action with
    * @throws CTPException if action state transition error
    */
-  private void transitionAction(final Action action, final ActionDTO.ActionEvent event)
-      throws CTPException {
-    final ActionDTO.ActionState nextState =
-        actionSvcStateTransitionManager.transition(action.getState(), event);
+  private void transitionAction(final Action action, final ActionDTO.ActionEvent event) {
+    ActionDTO.ActionState nextState = null;
+
+    try {
+      nextState = actionSvcStateTransitionManager.transition(action.getState(), event);
+    } catch (CTPException ctpExeption) {
+      throw new IllegalStateException();
```

Comment:
I'd prefer if aSSTM.transition() was changed to throw `IllegalStateException` rather than catching `CTPException` and throwing a more specific one

Reply:
yes I very much agree. It's one big refactor I want to do, to eliminate CTPException from the whole project - all 7+ Java 'micro' services do not need this checked exception and should not be using it like this.